gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/BloodPressureMapperTest.java

```diff
@@ -159,6 +184,47 @@ public void mapBloodPressureWithNoOptionalData() {
         assertThat(bloodPressure.getComponent().get(1).getReferenceRange()).isEmpty();
     }
 
+    @ParameterizedTest(name = "[{index}] {0}")
+    @ValueSource(strings = {
+        "ehr_composition_containing_confidentiality_code",
+        "battery_compound_statement_containing_confidentiality_code",
+        "systolic_observation_containing_confidentiality_code",
+        "diastolic_observation_containing_confidentiality_code"
+    })
+    public void When_MappingBloodPressureWithConfidentialityCodes_Expect_BloodPressureObservationContainsSecurityMeta(
+        String inputXml
+    ) {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Observation-1"),
+                confidentialityCodesCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtractElement(inputXml + ".xml");
+
+        final var bloodPressure = bloodPressureMapper
+            .mapResources(ehrExtract, patient, ENCOUNTER_LIST, PRACTISE_CODE)
+            .get(0);
+        final var confidentialityCode = confidentialityCodesCaptor
+            .getAllValues()
+            .get(0);
+        final var securityMeta = bloodPressure
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertThat(confidentialityCode).isPresent();
+        assertAll(
+            () -> assertThat(confidentialityCode.get())
+                .usingRecursiveComparison()
+                .isEqualTo(
+                    TestUtility.createCv(NOPAT_CODE, NOPAT_OID_CODESYSTEM, NOPAT_DISPLAY)
+                ),
```

Comment:
Given that you're only ever testing that NOPAT is passed into the `confidentialityService`

Could we simplify the production code to always pass in NOPAT? 😛